MetaMask/metamask-mobile
# fix: notification account syncing

## **Description**

This PR fixes a bug where account toggle were updated based on external events directly and re-rendered unnecessarily, while also preventing users to toggle multiple switches at once.

## **Related issues**

Fixes:

[NOTIFY-1112](https://consensyssoftware.atlassian.net/browse/NOTIFY-1112)
[NOTIFY-1111](https://consensyssoftware.atlassian.net/browse/NOTIFY-1111)

## **Manual testing steps**

1. Go to the notifications settings page.
2. Try to disable/enable multiple accounts at once. We expected you're blocked to do so.
3. Leave and go back to notifications settings page. State should be kept.
4. Import the same SRP Wallet in a different device and switch account on device A, while device B keeps settings screen open. We expect to NOT see the change propagated, unless device B leave and re-visits Notifications Settings screen.

## **Screenshots/Recordings**



### **Before**



### **After**

https://github.com/user-attachments/assets/ba72fdf0-b440-4ab2-b109-ee804dbced18




## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**



[NOTIFY-1112]: https://consensyssoftware.atlassian.net/browse/NOTIFY-1112?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[NOTIFY-1111]: https://consensyssoftware.atlassian.net/browse/NOTIFY-1111?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
